### PR TITLE
Pre release nightly  build and uploads

### LINKED ISSUE #86: Pre-release versions of coiled-runtime
It would be very helpful for us to have pre-releases of coiled-runtime available, either cut directly from dask `main` or a hand selected more recent version.

We could use this internally at coiled on our staging deployment as well to test things before promoting it to a full release

### PR DESCRIPTION
**Pull request by ncclementi**
- [x] Closes #86

### CODE REVIEW

**Comment by ncclementi** [comment]
So the new meta file now seems to work, but I'm still struggling to get the build going. I think it's a directory problems, and where things live. I wonder if saving the new meta.yml inside a directory in recipes/ makes it easier. Will try this next.

**Comment by ncclementi** [comment]
@shughes-uk visibility ping to let you know that when this PR gets merged you'll have pre-release nightly build available.

File: .github/workflows/pre-release.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+          # conda search for the latest distributed pre-release
+          arr_dist=($(conda search --override-channels -c dask/label/dev distributed | tail -n 1))
+          # extract distributed pre-release version / build
+          export DISTRIBUTED_VERSION=${arr_dist[1]}
+
+          # suffix for nightly package versions
+          export VERSION_SUFFIX=a`date +%y%m%d%H%M`
+
+          #build teh package
```
Comment (jrbourbeau) [inline .github/workflows/pre-release.yml:59]:
```suggestion
          # Build nightly `coiled-runtime` package
```

File: README.md [hunk 2]
```diff
@@ -12,6 +12,14 @@ The Coiled Runtime is a conda metapackage which makes it easy to get started wit
 conda install -c conda-forge coiled-runtime
 ```
 
+**Nightly builds**
+  
+If you are interested in installing a nightly version of the coiled-runtime:
+
+```bash
+conda install -c coiled/label/dev -c dask/label/dev  coiled-runtime
```
Comment (jrbourbeau) [inline README.md:20]:
```suggestion
conda install -c coiled/label/dev -c dask/label/dev coiled-runtime 
```

File: README.md [hunk 3]
```diff
@@ -12,6 +12,14 @@ The Coiled Runtime is a conda metapackage which makes it easy to get started wit
 conda install -c conda-forge coiled-runtime
 ```
 
+**Nightly builds**
+  
+If you are interested in installing a nightly version of the coiled-runtime:
```
Comment (jrbourbeau) [inline README.md:17]:
```suggestion
`coiled-runtime` has nightly conda packages for testing purposes.
You can install a nightly version of `coiled-runtime` with:
```

File: .github/workflows/pre-release.yml [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Set up python
+        uses: conda-incubator/setup-miniconda@v2
+        with:
+          miniforge-variant: Mambaforge
+          use-mamba: true
+          python-version: 3.9
+          environment-file: ci/environment.yml
+      - name: Install dependencies
+        run: |
+          mamba install boa conda-verify
```
Comment (jrbourbeau) [inline .github/workflows/pre-release.yml:40]: Is `conda-verify` needed?

**Review by jrbourbeau — COMMENTED** [review]
Thanks for your work on this @ncclementi

File: .github/workflows/pre-release.yml (hunk 4 above)
Comment (ncclementi) [reply .github/workflows/pre-release.yml:40]: I believe it is if you want to run the verification parts while the build happens. We do have it in dask and distributed.

**Review by jrbourbeau — APPROVED** [review]
Thanks @ncclementi, this is in

**Pull request merged**